backend/lookup/api/views.py

```diff
@@ -6,61 +6,62 @@
+    _visitors = _visitors_data.getVisitors() # initialize the visitors field(type: dict): {uuid: face_encodings}
+    __LAPSE = datetime.timedelta(seconds=40) # waiting time for a duplicate user
+    __TOLERANCE = 0.6 # threshold for face_distance
 
     def list(self, request):
         """Use this overridden method to list all the visitors in admin page."""
-        queryset = Visitor.objects.all().order_by("-created_at")
-        serializer = self.serializer_class(queryset, many=True)
-        return Response(serializer.data)
+        self._visitors_data.updateQueryset() # use this to update the queryset everytime any instance is deleted from the DB.
+        self._visitors_data.updateSerializer(save=False) # update the serializer with the queryset above
+        serializer = self._visitors_data.getSerializer() # get the serializer 
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
     def create(self, request):
         """Add request as a lookup instance after verication"""
-        LAPSE = datetime.timedelta(seconds=40) # waiting time for a duplicate user
-        queryset = Visitor.objects.all().order_by("-created_at")
-        serializer = self.serializer_class(queryset, many=True) # all the visitors in DB
-        users = dict(zip([data['id'] for data in serializer.data], [data['encoding'] for data in serializer.data])) # create a dictionary {uuid: encoding} of all the visitors
-
+        self._visitors_data.updateQueryset() # use this to update the queryset everytime any instance is deleted from the DB.
+        self._visitors_data.updateSerializer(save=False) # update the serializer with the queryset above
+        self._visitors_data.updateVisitors() # update the _visitors field with the updated serializer above
+        self._visitors = self._visitors_data.getVisitors()
         request.data.setlist('encoding', list(map(float, request.data.getlist('encoding')))) # convert encoding type from str to float(for calculation)
-        user_matched = face_recognition.compare_faces(list(users.values()), np.array(request.data.getlist('encoding'))) # calculate similarity between a requested user and all the visitors in DB(necessity of improvement in efficiency of calculation(redundant calculation?))
-        print('matched?', user_matched) # [True/False] -> If True is in the list, there's data of a current visitor
 
-        for i, user_matched in enumerate(user_matched): # Check if the same visitor exists
-            if user_matched: # if there's the same user
-                user_id_matched = list(users.keys())[i] # get the visitor's uuid
-                instance = self.queryset.get(pk=user_id_matched) # find an instance of the visitor
-                if timezone.now() - instance.recent_access_at <= LAPSE: # if the same user continuously tries to check in
+        if self._visitors: # If there're any existing visitors -> if false, the visitor is recorded as a new visitor automatically
```

Comment:
A new way to determine a new or existing visitor is implemented here(refer to the description of this PR.)